test/socket.jl

```diff
@@ -414,3 +414,15 @@ end
         @test 0 == ccall(:jl_tcp_reuseport, Int32, (Ptr{Void},), s.handle)
     end
 end
+
+@static if Sys.isapple()
+    @testset "Issues #18818 and #24169" begin
+        ulimit = readchomp(`ulimit -S -s`)
+        try
+            run(`ulimit -S -s 7001`)
```

Comment:
this will have no effect

Reply:
I'm not sure what you mean. Running that locally fails without this change and works with it.

Reply:
```
julia> readchomp(`ulimit -S -s`)
"8192"

julia> readchomp(`ulimit -S -s 7001`)
""

julia> readchomp(`ulimit -S -s`)
"8192"
```

This is a shell-builtin. So you either need to ccall the setrlimit, or spawn this test with a shell that can process both commands.

Reply:
Okay, I've reimplemented the test with `ccall`. Could you take another look, @vtjnash?